jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppClassLoader.java

```diff
@@ -325,10 +326,15 @@ private boolean isFileSupported(String file)
      */
     public void addJars(Resource lib)
     {
-        if (lib.exists() && lib.isDirectory())
+        if (lib.exists() && lib.isDirectory()) // TODO perhaps redundant given null check from list()?
         {
             String[] files=lib.list();
-            for (int f=0;files!=null && f<files.length;f++)
+            if (files == null)
+            {
+                return;
+            }
+            Arrays.sort(files);
```

Comment:
Could of course use any deterministic `Comparator` if there is some reason not to want lexicographic.

Reply:
This smells like it will hit its head against the servlet spec descriptor ordering requirements.  (Not 100% sure)

Reply:
Presumably if there were some requirement that something be ordered in a particular way, you were already violating that spec, since the current behavior was to pick up the JARs in a random order.

Reply:
The servlet spec descriptor ordering is only for the discovery of web fragments and resources  within the jars.  It does not apply to classloading as duplicate classes are not considered.